src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +22,35 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    private static final int NUMBER_OF_CHARS = 5;
+
+    public String getUniqueCharacters(String fileName) throws IOException {
+        TreeSet<Character> charSet = new TreeSet<>();
+        String finalChars = "";
+
+        try (BufferedReader fileReader = new BufferedReader(new FileReader(fileName))) {
+            while (fileReader.ready()) {
+                String charsOnly = fileReader.readLine().toLowerCase().replaceAll("[^a-zA-Z]", "");
+                for (int i = 0; i < charsOnly.length(); i++) {
+                    char currentChar = charsOnly.charAt(i);
+                    if (!charSet.contains(currentChar)) {
+                        charSet.add(currentChar);
+                    }
+                }
+            }
+
+            int size = charSet.size();
```

Comment:
try to simplify this part of the method

Reply:
fixed

Reply:
how many times will the loop run if charSet contains all 26 characters? Can you think of way to reduce number of iterations in this case?

This comment is optional.